### DISCUSSION TITLE
unique database column value validation

### DISCUSSION THREAD

**Discussion by heyjoe1984** [blitz-js/blitz#2982] [Q&A]
how do I validate that a value is unique (does not exist yet) in the prisma database?

I checked the validations for signup and didn't see if it was checking for this

I couldn't find anything in the docs either, maybe I am missing something

**Comment by beerose** [comment]
If you want to do the validation before submitting a form (ie before calling a corresponding mutation), you can use `refine` from `zod` that lets you perform some async logic like calling the database to check some information: https://github.com/colinhacks/zod#asynchronous-refinements.

However, if you have a `unique` constraint on this field in the database, then the database will throw an error if you try to create the same entry. Then, you can check the error code, and if it's `P2002` (docs: https://www.prisma.io/docs/reference/api-reference/error-reference#p2002), that means the unique constraint was violated.
